skiko/src/awtMain/kotlin/org/jetbrains/skiko/redrawer/MetalRedrawer.kt

```diff
@@ -133,10 +165,13 @@ internal class MetalRedrawer(
         // When window is not visible - it doesn't make sense to redraw fast to avoid battery drain.
         // In theory, we could be more precise, and just suspend rendering in
         // `NSWindowDidChangeOcclusionStateNotification`, but current approach seems to work as well in practise.
-        if (isOccluded(windowHandle))
+        if (isOccluded())
             delay(300)
     }
 
+    /**
+     * Avoid calling this method on main thread, it will sleep before next vsync happens if invoked multiple times.
```

Comment:
It's a bit different thing:

`setVSyncEnabled(device.ptr, enabled = false)` is related [to this](https://developer.apple.com/documentation/quartzcore/cametallayer/2887087-displaysyncenabled?language=objc)

It affects internal implementation of how macOS window synchronizes blit operations of _CAMetalDrawable_ content into _IOSurface_.  Logic added by me here is related to VSync-based timer for drawing stuff to avoid presenting more than one drawable per vsync.